ProteGO/Common/DangerStatusManager/DangerStatusManager.swift

```diff
@@ -2,29 +2,45 @@ import Foundation
 import SwiftTweaks
 import RxSwift
 import RxCocoa
+import Valet
 
 final class DangerStatusManager: DangerStatusManagerType {
+
     var currentStatus: BehaviorRelay<DangerStatus>
 
     private var tweakBindings = Set<TweakBindingIdentifier>()
 
-    init() {
-        self.currentStatus = BehaviorRelay<DangerStatus>(value: .yellow)
+    private let gcpClient: GcpClientType
 
-        if DebugMenu.assign(DebugMenu.forceDangerStatus) {
-            if let status = DangerStatus.init(rawValue: DebugMenu.assign(DebugMenu.forceDangerStatusValue).value) {
-                self.currentStatus.accept(status)
-            }
+    private let valet: Valet
+
+    private let disposeBag = DisposeBag()
+
+    init(gcpClient: GcpClientType, valet: Valet) {
+        self.gcpClient = gcpClient
+        self.valet = valet
 
-            tweakBindings.insert(DebugMenu.bind(DebugMenu.forceDangerStatusValue) { value in
-                if let status = DangerStatus.init(rawValue: value.value) {
-                    self.currentStatus.accept(status)
-                }
-            })
-        }
+        let initialDangerStatus = valet.string(forKey: Constants.KeychainKeys.currentDangerStatus)
+            .flatMap(DangerStatus.init(rawValue:)) ?? .yellow
+
+        self.currentStatus = BehaviorRelay<DangerStatus>(value: initialDangerStatus)
     }
 
-    deinit {
-        self.tweakBindings.forEach(DebugMenu.unbind)
+    func updateCurrentDangerStatus() {
```

Comment:
Myślałeś o tym, żeby subskrybować w VC albo przynajmniej wystawić Observable'a informującego o tym czy wykonywana jest operacja? Coś typu 'isLoading: Observable<Bool>` żeby można było pokazać spinner podczas uaktualniania statusu

Reply:
dobry pomysł, zapytam o do designerów i dodamy to najwyzej w oddzielnym PR

Reply:
(bo teraz nie mamy designów teraz na pokazywanie takeigo spinnera czy progressu)